test/test_parser.rb

```diff
@@ -10107,6 +10107,73 @@ def test_endless_setter
       SINCE_3_0)
   end
 
+  def test_endless_comparison_method
+    assert_parses(
```

Comment:
could you rewrite it into a loop that iterates over method names please? Also, source maps are redundant here.